File: python/gemini/p124_fc.py
import sys

def main():
    if len(sys.argv) != 2:
        print("Error: Exactly one argument required.", file=sys.stderr)
        sys.exit(1)
    
    input_val = sys.argv[1]
    print(f"Original: '{input_val}'")
    
    trimmed = input_val.rstrip()
    print(f"Trimmed: '{trimmed}'")
    
    sys.exit(0)

if __name__ == "__main__":
    main()

File: c/gemini/p124_fc.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(int argc, char *argv[]) {
    if (argc != 2) {
        fprintf(stderr, "Error: Exactly one argument required.\n");
        return 1;
    }

    char *input = argv[1];
    printf("Original: '%s'\n", input);

    size_t len = strlen(input);
    while (len > 0 && isspace((unsigned char)input[len - 1])) {
        input[--len] = '\0';
    }

    printf("Trimmed: '%s'\n", input);
    return 0;
}